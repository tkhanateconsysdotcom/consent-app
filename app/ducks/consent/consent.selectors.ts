import { pathOr } from "ramda";

import { namespace } from "./consent.slice";

const getMonthlyAcceptanceConsent = (state) =>
  pathOr(0, [namespace, "monthlyAcceptanceConsent"], state);

export default {
  getMonthlyAcceptanceConsent,
};
