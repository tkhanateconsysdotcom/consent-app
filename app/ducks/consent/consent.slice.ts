import { createSlice, PayloadAction } from "@reduxjs/toolkit";

const initialState = {
  monthlyAcceptanceConsent: 166,
};

const partners = createSlice({
  name: "consent",
  initialState,
  reducers: {
    setMonthlyAcceptanceConsent(state, action: PayloadAction<number>) {
      state.monthlyAcceptanceConsent = action.payload;
    },
  },
});

const { setMonthlyAcceptanceConsent } = partners.actions;

export const actions = { setMonthlyAcceptanceConsent };

export const namespace = partners.name;

export const reducer = partners.reducer;
