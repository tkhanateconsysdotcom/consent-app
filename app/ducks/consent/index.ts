import { actions, namespace, reducer } from "./consent.slice";
import selectors from "./consent.selectors";

export default { actions, namespace, reducer, selectors };
