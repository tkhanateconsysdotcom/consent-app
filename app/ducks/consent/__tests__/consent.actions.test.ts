import consent from "../";

describe("Consent actions should", () => {
  it("export expected actions", () =>
    expect(consent.actions).toMatchSnapshot());

  it("be able to set a monthly consent", () => {
    const monthlyAcceptanceConsent = 200;

    const expectedAction = {
      type: consent.actions.setMonthlyAcceptanceConsent.toString(),
      payload: monthlyAcceptanceConsent,
    };
    expect(
      consent.actions.setMonthlyAcceptanceConsent(monthlyAcceptanceConsent)
    ).toEqual(expectedAction);
  });
});
