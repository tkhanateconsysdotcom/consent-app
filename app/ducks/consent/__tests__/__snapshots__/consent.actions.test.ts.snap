// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consent actions should export expected actions 1`] = `
Object {
  "setMonthlyAcceptanceConsent": [Function],
}
`;
